src/Npgsql/NpgsqlRawCopyStream.cs

```diff
@@ -73,36 +74,48 @@ internal NpgsqlRawCopyStream(NpgsqlConnector connector)
         _copyLogger = connector.LoggingConfiguration.CopyLogger;
     }
 
-    internal async Task Init(string copyCommand, bool async, CancellationToken cancellationToken = default)
+    internal async Task Init(string copyCommand, CopyOperationType copyOperationType, bool async, CancellationToken cancellationToken = default)
     {
-        await _connector.WriteQuery(copyCommand, async, cancellationToken).ConfigureAwait(false);
-        await _connector.Flush(async, cancellationToken).ConfigureAwait(false);
+        TraceCopyStart(copyCommand, copyOperationType);
 
-        using var registration = _connector.StartNestedCancellableOperation(cancellationToken, attemptPgCancellation: false);
+        try
+        {
+            await _connector.WriteQuery(copyCommand, async, cancellationToken).ConfigureAwait(false);
+            await _connector.Flush(async, cancellationToken).ConfigureAwait(false);
 
-        var msg = await _connector.ReadMessage(async).ConfigureAwait(false);
-        switch (msg.Code)
+            using var registration = _connector.StartNestedCancellableOperation(cancellationToken, attemptPgCancellation: false);
+
+            var msg = await _connector.ReadMessage(async).ConfigureAwait(false);
+            switch (msg.Code)
+            {
+            case BackendMessageCode.CopyInResponse:
+                _state = CopyStreamState.Ready;
+                var copyInResponse = (CopyInResponseMessage)msg;
+                IsBinary = copyInResponse.IsBinary;
+                _canWrite = true;
+                _writeBuf.StartCopyMode();
+                TraceSetImport();
```

Comment:
Just confirming about this late setting of the operation after the activity/span has already been started above: is this confirmed to be OK? Like will the tag be sent with the closed/exception event and be taken into account?

Reply:
Can confirm, at the moment when an activity is disposed, all the tags it has will be sent, no matter at which point a specific tag is added. The only difference is that you can (potentially) filter new activities, for example disabling if it has a specific tag. But that's not something we should worry right now.